lib/iris/_concatenate.py

```diff
@@ -431,9 +431,8 @@ def match(self, other, error_on_mismatch):
 
         # Check if either cube is anonymous.
         if self.anonymous or other.anonymous:
-            msg = ('Dimensions differ: one or both cubes have anonymous '
-                   'dimensions')
-            msgs.append(msg)
+            msg = ('One or both cubes have anonymous dimensions')
```

Comment:
Its perfectly valid to have anonymous dimensions and since concatenation now handles them so why issue the warning?
Is iris making an assumption under these circumstances which might lead it to doing the 'wrong' thing, hence pointing it out to the user in the form of a warning?

In our application that we provide to our users, we currently have to filter certain unnecessary warnings issues by certain libraries so as to be left with those which the user does need be aware of.  Is there a purpose to this warning that I'm missing?

Cheers